### DISCUSSION TITLE
How to use Grafana Loki for log reporting??

### DISCUSSION THREAD

**Discussion by vincenzopalazzo** [open-telemetry/opentelemetry-rust#1469] [Q&A] [answered]
I am starting to look inside the [opentelemetry-rust](https://github.com/open-telemetry/opentelemetry-rust) and I was looking for a way to report log in Grafana Loki

I am not able to find a way with [opentelemetry-appender-log](https://github.com/open-telemetry/opentelemetry-rust/tree/main/opentelemetry-appender-log) so I was thinking that maybe 
Grafana Loki is not supported?

There are any hints on how to do it? thanks

**Comment by lalitb** [comment, marked as answer]
There is a [Loki exporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/lokiexporter) for OTLP Collector, so you should be able to configure OTLP collector to push logs to loki as below:

`logs` -> `opentelemetry-appender-log` -> `OTLP exporter` -> `OTLP Collector` -> `Loki Exporter` -> `Loki`

**Reply by vincenzopalazzo** [reply]
Thanks, I am not able to see these steps `OTLP exporter -> OTLP Collector -> Loki Exporter -> Loki` in any examples

**Reply by lalitb** [reply]
The example of exporting logs to OTLP Collector - https://github.com/open-telemetry/opentelemetry-rust/tree/main/opentelemetry-otlp/examples/basic-otlp, so this covers the first half of the pipeline:
`logs` -> `opentelemetry-appender-log` -> `OTLP exporter` -> `OTLP Collector`

Further, the example for exporting logs from OTLP Collector to Loki - https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/lokiexporter/example/otelcol.yaml, so this covers the later half of pipeline:
`OTLP Collector` -> `Loki exporter` -> `Loki`

**Reply by vincenzopalazzo** [reply]
Ok I got it thanks.

I had another question regarding the usability of this: Why I can not do the following process `logs -> opentelemetry-appender-log -> Loki exporter -> Loki`? and I should pass through the `OTLP exporter`?

**Reply by lalitb** [reply]
There is no native Loki exporter built in Rust which you can use, to the best of my knowledge.  At-least, we don't have anything such in otel-rust and otel-rust-contrib repo. So the solution is either of these
 - Implement and contribute the Loki exporter in Rust.
 - Use the OTLP collector's Loki exporter.
